holopy/core/io/io.py

```diff
@@ -151,13 +153,77 @@ def save(outf, obj):
     binary array is very slow for large arrays.  HoloPy can read these 'yaml'
     files, but any other yaml implementation will get confused.
     """
+    #TODO: Default to saving hdf5, with option to save yaml
+    #TODO: can't save images with this function
     if isinstance(outf, str):
         filename, ext = os.path.splitext(outf)
-        if ext in ['.tif', '.TIF', '.tiff', '.TIFF']:
+        if ext in tiflist:
             save_image(outf, obj)
             return
     serialize.save(outf, obj)
 
+def save_image(filename, im, scaling='auto', depth=8):
+    """Save an ndarray or image as a tiff.
+
+    Parameters
+    ----------
+    im : ndarray or :class:`holopy.image.Image`
+        image to save.
+    filename : basestring
+        filename in which to save image. If im is an image the
+        function should default to the image's name field if no
+        filename is specified
+    scaling : 'auto', None, or (None|Int, None|Int)
+        How the image should be scaled for saving. Ignored for float
+        output. It defaults to auto, use the full range of the output
+        format. Other options are None, meaning no scaling, or a pair
+        of integers specifying the values which should be set to the
+        maximum and minimum values of the image format.
+    depth : 8, 16 or 'float'
+        What type of image to save. Options other than 8bit may not be supported
+        for many image types. You probably don't want to save 8bit images without
+        some kind of scaling.
+
+    """
+    # if we don't have an extension, default to tif
+    if os.path.splitext(filename)[1] is '':
+        filename += '.tif'
+
+    if scaling is not None:
+        if scaling is 'auto':
+            min = im.min()
+            max = im.max()
+        elif len(scaling) == 2:
+            min, max = scaling
+        else:
+            raise Error("Invalid image scaling")
+        if min is not None:
+            im = im - min
+        if max is not None:
+            im = im / (max-min)
+
+    if depth is not 'float':
+        if depth is 8:
+            depth = 8
+            typestr = 'uint8'
+        elif depth is 16 or depth is 32:
+            depth = depth-1
+            typestr = 'int' + str(depth)
+        else:
+            raise Error("Unknown image depth")
+            
+        if im.max() <= 1:
+            im = im * ((2**depth)-1) + .499999
+            im = im.astype(typestr)
+    if os.path.splitext(filename)[1] in tiflist:
+        metadat = yaml.dump(im._dict)
```

Comment:
I think we may want to look into using json instead of yaml for metadata like this. Don't need to fix that before merging this code though.